packages/api/tests/endpoints/test-granules.js

```diff
@@ -613,42 +749,24 @@ test('DELETE deleting an existing unpublished granule', async (t) => {
   const { detail } = response.body;
   t.is(detail, 'Record deleted');
 
-  // verify the files are deleted
+  // verify the files are deleted from S3. No need to check the Postgres files.
+  // If the granule was successfully deleted, the postgres
+  // files will have been as well. Files have a fk which would
+  // prevent the granule from being deleted if files referencing it
+  // still exist.
   /* eslint-disable no-await-in-loop */
   for (let i = 0; i < newGranule.files.length; i += 1) {
     const file = newGranule.files[i];
-    t.false(await fileExists(file.bucket, file.key));
+    t.false(await s3ObjectExists({ Bucket: file.bucket, Key: file.key }));
   }
   /* eslint-enable no-await-in-loop */
 
-  await deleteBuckets([
-    buckets.protected.name,
-    buckets.public.name,
+  await deleteS3Buckets([
+    s3Buckets.protected.name,
+    s3Buckets.public.name,
```

Comment:
If this is test cleanup, which I think it is, you should put it in a `t.teardown` call, otherwise it only runs if the test doesn't fail before it reaches this line.

```suggestion
  t.teardown(() => deleteS3Buckets([
    s3Buckets.protected.name,
    s3Buckets.public.name,
  ]));  
```

Reply:
Ah! Ok, I didn't realize. I updated two tests that create S3 buckets here:

614387690197bef7a50fe999b128b090e0f02eab

I'm also running those two in serial now because I think there's an issue where, if we run them in parallel, one can delete the buckets that the other then tries to delete and fails. Apparently `teardown` doesn't work in an `after` hook (which I guess makes sense) so this might be the best way to clean up?

Reply:
Yeah `teardown` is meant to be called directly inside the test where the setup happens